tests/system/kibana.py

```diff
@@ -0,0 +1,82 @@
+#import json
+from urllib.parse import urljoin
+
+import requests
+
+class Kibana(object):
+    def __init__(self, url):
+        self.url = url
+
+    def create_agent_config(self, service, settings, agent=None, env=None):
+        resp = self._upsert_agent_config(service, settings, agent=agent, env=env)
+        result = resp.json()["result"]
+        assert result == "created", result
+        return resp
+
+    def create_or_update_agent_config(self, service, settings, agent=None, env=None):
+        resp = self._upsert_agent_config(service, settings, agent=agent, env=env, overwrite=True)
+        result = resp.json()["result"]
+        assert result in ("created", "updated"), result
+        return resp
+
+    def _upsert_agent_config(self, service, settings, agent=None, env=None, overwrite=False):
```

Comment:
Not sure if this `_` is an oversight or on purpose, but in case it is used to signal internal usage only, please do the same for `list_agent_config` and `delete_agent_config` and group them together. 
IMO we should just remove the underscore here though.

Reply:
`_` is copied across from test_integration_acm, and is intended to signal internal usage; it exists only to avoid repeating code in the above callers

I think `list_agent_config` and `delete_agent_config` could be useful outside of this class, as they're part of the ACM API. They're not used _yet_, but I don't see harm in exposing them.